src/behavior-considered-undefined.md

```diff
@@ -23,44 +23,58 @@ code.
 </div>
 
 * Data races.
-* Dereferencing a null or dangling raw pointer.
-* Unaligned pointer reading and writing outside of [`read_unaligned`]
-  and [`write_unaligned`].
-* Reads of [undef] \(uninitialized) memory.
-* Breaking the [pointer aliasing rules] on accesses through raw pointers;
-  a subset of the rules used by C.
-* `&mut T` and `&T` follow LLVM’s scoped [noalias] model, except if the `&T`
-  contains an [`UnsafeCell<U>`].
-* Mutating non-mutable data &mdash; that is, data reached through a shared
-  reference or data owned by a `let` binding), unless that data is contained
+* Dereferencing (using the `*` operator on) a dangling or unaligned raw pointer.
+* Breaking the [pointer aliasing rules]. `&mut T` and `&T` follow LLVM’s scoped
+  [noalias] model, except if the `&T` contains an [`UnsafeCell<U>`].
+* Mutating non-mutable data (that is, data reached through a shared
+  reference or data owned by an immutable binding), unless that data is contained
   within an [`UnsafeCell<U>`].
-* Invoking undefined behavior via compiler intrinsics:
-  * Indexing outside of the bounds of an object with [`offset`] with
-    the exception of one byte past the end of the object.
-  * Using [`std::ptr::copy_nonoverlapping_memory`], a.k.a. the `memcpy32`and
-    `memcpy64` intrinsics, on overlapping buffers.
-* Invalid values in primitive types, even in private fields and locals:
-  * Dangling or null references and boxes.
+* Invoking undefined behavior via compiler intrinsics.
+* Executing code compiled with platform features that the current platform
+  does not support (see [`target_feature`]).
+* Unwinding into another language.
```

Comment:
I think this would be better as something like this:

```suggestion
* Unwinding into another language or into Rust compiled with a different toolchain version or into Rust compiled with the same toolchain version but different compilation options except for those options for which we note otherwise.
```

Reply:
All of this "toolchain" stuff is highly implementation (rustc) specific and falls under "the ABI is not stable".

Reply:
@Centril do you have a better way to word this? `"Unwinding into another language"` is UB, but so is unwinding into Rust in many cases as well.

Reply:
Can we replace that list of "or <different Rust>" by something like "or into a separately compiled Rust library"? Not sure what the best wording is, I am looking for something that basically days "if you link together two Rust dylib/cdylib, that's basically crossing a language barrier".

Reply:
Maybe "Violating the call ABI and unwinding ABI of a function on function calls" ?

Reply:
Another example is that, AFAICT, according to this list, `transmuting` an `extern "Rust" fn` to an `extern "C" fn` is not UB, and I can't find any item that says that calling the Rust function via the `extern "C" fn` is UB. 

A bullet point saying that "Calling a function with an incorrect call ABI (including unwinding ABI) is UB" would probably solve both issues.

Reply:
I made it "Calling a function with the wrong call ABI (including unwind ABI)."